src/test/preprocessing/preprocessing-modules/Validator.test.ts

```diff
@@ -0,0 +1,65 @@
+import { module as Validator } from '../../../preprocessing/preprocessing-modules/Validator';
+import { JSDOM } from 'jsdom';
+
+declare global {
+  namespace NodeJS {
+    interface Global {
+      window: Window;
+      document: Document;
+    }
+  }
+}
+
+describe('Validator module', () => {
+  // tslint:disable:no-any
+  const config = {} as any;
+
+  beforeAll(() => {
+    const dom = new JSDOM('', {
+      runScripts: 'dangerously',
+      resources: 'usable',
+    });
+    global.window = dom.window;
+    global.document = dom.window.document;
+  });
+
+  afterAll(() => {
+    delete global.window;
+    delete global.document;
+  });
+
+  test('has correct name', () => {
+    expect(Validator.name).toBe('Validator');
+  });
+
+  test('works on valid AMP', async () => {
+    const code = `<!doctype html>
+    <html ⚡4email>
+    <head>
+      <meta charset="utf-8">
+      <script async src="https://cdn.ampproject.org/v0.js"></script>
+      <style amp4email-boilerplate>body{visibility:hidden}</style>
+    </head>
+    <body>Hello</body>
+    </html>`;
+
+    const out = await Validator.process(code, config);
+    expect(out).toBe(code);
+  });
+
+  test('throws on invalid AMP', async () => {
+    const code = `<!doctype html>
+    <html ⚡4email>
```

Comment:
can we also add a test to make sure it throws when validating a different AMP spec but valid AMP doc, e.g  just vanilla AMP , testing that we are validating against the amp4email spec.
`<html ⚡>
 `

Reply:
Good idea, added.